blocks/hooks/custom-classname.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * External dependencies
+ */
+import { assign } from 'lodash';
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { hasBlockSupport } from '../api';
+import InspectorControls from '../inspector-controls';
+import { extendElement } from './utils';
+
+/**
+ * Filters registered block settings, extending attributes with anchor using ID
+ * of the first node
+ *
+ * @param  {Object} settings Original block settings
+ * @return {Object}          Filtered block settings
+ */
+export function addAttribute( settings ) {
+	if ( hasBlockSupport( settings, 'customClassName', true ) ) {
+		// Use Lodash's assign to gracefully handle if attributes are undefined
+		settings.attributes = assign( settings.attributes, {
```

Comment:
@aduth is there any particular reason why the existing variable is mutated here? I think @youknowriad is following your implementation. It looks like a reason why `let` needs to be used later in unit tests: https://github.com/WordPress/gutenberg/pull/3472/files#diff-1f4671e20828b625e6bfcb26ec4c7cdeR19. It would be nice to make it immutable, but maybe I'm missing something.

Reply:
https://github.com/WordPress/gutenberg/pull/3318#discussion_r148728475

Reply:
Yes, thank. I  discussed it with Andrew, too. It makes the most sense in the context of `getSaveContent.extraProps`. Let's leave it as it is.